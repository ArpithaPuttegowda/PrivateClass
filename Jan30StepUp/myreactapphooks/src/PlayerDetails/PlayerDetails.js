import React from "react";

export const PlayerDetails = ({ name, loc }) => {
  return (
    <div>
      <b>Name:</b>
      <span>{name}</span>
      <b>Loc:</b>
      <span>{loc}</span>
    </div>
  );
};
