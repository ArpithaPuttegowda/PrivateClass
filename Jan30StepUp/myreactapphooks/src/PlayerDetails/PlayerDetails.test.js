import { render, screen } from "@testing-library/react";
import { PlayerDetails } from "./PlayerDetails";

describe("PlayerDetails", () => {
  it("should render successufully", () => {
    const { container } = render(<PlayerDetails />);
    expect(container).toMatchSnapshot();
  });
  it("should render the player name with location", () => {
    render(<PlayerDetails name="Sachin" loc="Mumbai" />);
    expect(screen.getByText("Name:")).toBeInTheDocument();
    expect(screen.getByText("Sachin")).toBeInTheDocument();
    expect(screen.getByText("Loc:")).toBeInTheDocument();
    expect(screen.getByText("Mumbai")).toBeInTheDocument();
  });
  it("should render the player name for different props", () => {
    render(<PlayerDetails name="Kohli" loc="Delhi" />);
    expect(screen.getByText("Kohli")).toBeInTheDocument();
    expect(screen.getByText("Delhi")).toBeInTheDocument();
  });
});
