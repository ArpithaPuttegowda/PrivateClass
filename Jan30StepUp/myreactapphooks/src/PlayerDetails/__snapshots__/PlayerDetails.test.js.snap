// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerDetails should render successufully 1`] = `
<div>
  <div>
    <b>
      Name:
    </b>
    <span />
    <b>
      Loc:
    </b>
    <span />
  </div>
</div>
`;
